### DISCUSSION TITLE
Server crash detection methods?

### DISCUSSION THREAD

**Discussion by eldorel** [bitfocus/companion#2556] [Q&A]
Q: Is there any way to have a surface indicate that the server has gone down? 

We had a power cut last night and the box running companion didn't restart. 

I happen to be using the 'web buttons' UI on a handful of tablet devices with batteries, so the web UI on them remained loaded the entire time. 

I know there are quite a few external options for service uptime monitoring and reporting, but I'l like for the tablets themselves to reflect the error so that people who don't get that notification aren't pressing buttons and wondering why it doesn't work. 

I've figured out how to have the buttons all turn red when the device they control goes down, and I'm hoping that there's a way to have a similar function when companion is dead.

**Comment by kraves-oz** [comment]
You could try checking the internal variable that reports IP address, reading it off a button text or something similar. Might be a more direct method though.

**Comment by Julusian** [comment]
I suppose the correct thing to do here would be to make the tablet and emulator pages show connection status properly. That would probably solve your issue?
